### ISSUE TITLE
On pickup event: event-player bug

### ISSUE THREAD

**Issue by Guarmanda** [SkriptLang/Skript#1322]
this code will send a message to the player

on pickup:
    message "%event-location%" to event-player
    cancel event

this one won't:

on pickup:
    message "%event-location%"
    cancel event

**Comment by TheBentoBox** [comment]
The issue here is that if you send a message without specifying the recipient, Skript automatically sends it to the `event-entity`, which in this event is the dropped item entity which you are picking up. I believe there have been issues opened that were somewhat related to this in the past, but I can't find an open one; if someone else does, feel free to link it for me pls.

Ideally, it would make more sense for the event to try to "find a player" before just finding whatever entity it can (something like a try catch around `event.getPlayer()`, but using try/catch for logic flow is gross). That being the case I'm not really sure if there's a great way to fix this or if it's something the devs even want to fix, so I'll leave that for them to decide.

**Comment by Snow-Pyon** [comment]
Since all entities are command senders, Skript takes the first entity it finds from the event, this could be improved though (i.e. prioritize players when it fills the placeholders).

**Comment by TheLimeGlass** [comment]
I'm interested in this, I don't know much about this and I want to know how and where Skript determines where the entity/player comes from. Like for example I just found out that the permission check condition doesn't even require a player, just one involved in the event/command. So like
```
has permission "example.neat":
```
How does Skript know about the player? I looked around but gave up after getting into the parser stuff.
